source/common/router/router.h

```diff
@@ -129,7 +129,9 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     if (route_entry_ && downstream_headers_) {
       auto hash_policy = route_entry_->hashPolicy();
       if (hash_policy) {
-        return hash_policy->generateHash(*downstream_headers_);
+        const Network::Connection* connection = this->downstreamConnection();
```

Comment:
In what situations do we now have a `downstreamConnection()`? Should this a be an `ASSERT` and/or reference instead?

Reply:
IIRC AsyncClient does not have a downstream connection. This actually confused me also and I had to think about it so I think worth a comment.

Reply:
also nit: in general we don't use `this->` in this type of context.